src/core/container_async.rs

```diff
@@ -0,0 +1,62 @@
+use crate::core::{DockerAsync, ImageAsync};
+
+/// Represents a running docker container using async trait.
+///
+/// Containers have a [`custom destructor`][drop_impl] that removes them as soon as they go out of scope:
```

Comment:
This might actually be a problem because Rust doesn't have async drop yet :(

I think we might need to hack around this by blocking the thread in the `Drop` impl. Not sure what the current best practice here is.

Reply:
Unfortunately after extensive reading and researching it's still a really hard problem and not even sure if rust wants to support it. I could do something crazy but rust async is designed to block people from doing such crazy thing. Like doing a `block_on` inside a Runtime context. I've tried a few and no luck.

Any suggestion on which route we should take? Is there a way to design the code around needing the `Drop`?

**The** article: https://boats.gitlab.io/blog/post/poll-drop/
Discussion:
- https://users.rust-lang.org/t/what-is-the-idiomatic-way-of-cleaning-up-async-resources-in-rust/48878
- https://stackoverflow.com/questions/59940976/panic-running-async-code-while-dropping-rust-future